github-git/2-identify-issue.md

```diff
@@ -0,0 +1,121 @@
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.16.4
+kernelspec:
+  display_name: Python 3 (ipykernel)
+  language: python
+  name: python3
+---
+
+(identify-issue)=
+# Identify an issue to work on 
+
+Open issues are where developers track bugs, new features, or other improvements needed in the repo. Whether you have a specific fix in mind or are looking for ideas, you should start by exploring existing open issues.
+
+In this lesson you will create or comment on an issue that you'd like to fix in GitHub repository that you don't own. 
+
+:::{admonition} Activity: Create (or find) an issue for a bug that you'd like to fix 
+
+1. Navigate to the [pyOpenSci example repo](https://github.com/pyOpenSci/pyos-demo-package-contribute). 
+2. Explore the documentation and code. You will find many items that need to be fixed, including spelling errors, typos, and more.
+3. Pick a file that you'd like to work on. Open a new issue specifying what you'd like to fix.
+4. Wait for someone on the pyOpenSci team to respond to you with next steps. 
+
+Once you have submitted an issue and someone has responded positively, you can begin working on the changes in your fork. [You will learn how to fork a repo, next.](fork-repo)
+:::
+
+As you scan the issues, look for beginner-friendly labels. Beginner-friendly labels are a clear message that the project welcomes new contributors. Many projects label issues as **good first issue** or **beginner-friendly** to highlight approachable tasks that are approachable for new contributors.
+
+## Open or comment on an issue
+  
+If you **have an idea already of what you'd like to work on**, check if an issue exists for it. If it does, comment on the issue to indicate your interest. If not, create a new issue and wait for feedback.
+
+Your comment might look like this:
+
+> Hey maintainer team 👋. I'm interested in working on this issue. Is there anything specific that I should consider before getting started?
+
+If you’re **looking for something to work on,** browse the issues to find one that interests you and comment on it to let maintainers know that you'd like to help. 
+
+:::{tip}
+Leaving comments in a new repository can feel intimidating, but most maintainers appreciate respectful communication and enthusiasm. If a project isn’t open to contributions, that’s okay—there are plenty of welcoming projects to explore!
+:::
+
+(create-issue)=
+### How to create a new issue
+
+:::{tip}
+Some projects use individual templates for bugs, documentation fixes, and new feature requests. Other projects might have no templates.
+:::
+
+If an issue doesn’t already exist for the thing you'd like to work on, here’s how to create a new one: 
+
+1. Click on the new issues button. If there is a template that you can select for the issue, use one.
+
+1. Write a **clear title** summarizing what you’d like to fix. Some repositories use templates—fill them out if available.
```

Comment:
some examples would probably be good here. 

bugs:
* bad: "program doesn't work"
* bad: "{copying a whole raw traceback}"
* good: "{error} when {condition}" - "OSError when passing relative path to `some_function`"

feature requests:
* bad: "Some suggestions" - too vague
* bad: "Improving the user experience by adding an extra parameter "relative=True/False" to `some_function` to allow the user to specify whether a path is relative or absolute" - too chatty
* good: "Add `relative` parameter to `some_function`"

Reply:
Aha, i see this is down below on line 91 and company. Leaving this here just to say "reading this as a newbie, i am a little confused at what is meant by a clear title, and that might be a bit distracting"

Reply:
Good catch! i moved it up to here! And put it in an admonition with a table :) 
<img width="861" alt="Screenshot 2025-01-30 at 6 25 25 PM" src="https://github.com/user-attachments/assets/9826b78f-3bbb-4927-8295-9d4d26c53d12" />